lib/tiered_category_expressions/expression.rb

```diff
@@ -37,23 +37,28 @@ def self.parse(str)
     end
 
     # @!visibility private
-    def initialize(tiers)
+    def initialize(tiers, strict:)
```

Comment:
I'd suggest to document the `strict` parameter. From the code I understand it matches less, but why do you need it?